sycl/test-e2e/format.py

```diff
@@ -87,19 +162,56 @@ def parseTestScript(self, test):
 
         return script
 
-    def getMatchedFromList(self, features, alist):
+    def getMissingRequiredFeaturesFromList(self, features, requires, build_only_mode = False):
         try:
             return [
-                item for item in alist if BooleanExpression.evaluate(item, features)
+                item
+                for item in requires
+                if not E2EExpr.evaluate(item, features, build_only_mode)
+            ]
+        except ValueError as e:
+            raise ValueError("Error in REQUIRES list:\n%s" % str(e))
+
+    def getMatchedFromList(self, features, alist, build_only_mode = False):
+        try:
+            return [
+                item for item in alist if E2EExpr.evaluate(item, features, build_only_mode, False)
             ]
         except ValueError as e:
             raise ValueError("Error in UNSUPPORTED list:\n%s" % str(e))
 
+    def select_triples_for_test(self, test):
+        supported_triples = set()
+        for t in test.config.available_triples:
+            features = test.config.available_features.union({t})
+            if self.getMissingRequiredFeaturesFromList(features, test.requires, True):
+                continue
+            if self.getMatchedFromList(features, test.unsupported, True):
+                continue
+            supported_triples.add(t)
+
+        if len(supported_triples) <= 1:
+            return supported_triples
+        # Treat XFAIL as UNSUPPORTED if the test is to be compiled for multiple
+        # triples.
+        if "*" in test.xfails:
```

Comment:
kinda unrelated question, i know we do the same today for multiple target devices but does it seem like it has to be this way or do it seem like it can be fixed if we put effort in to (which shouldnt be done as part of this pr obv)

Reply:
I think we're closer to being able to fix it now, so worth another look. @uditagarwal97 and I had talked about that recently. IMO, we should finish these "split" activities first so that changes won't have too many conflicts (both source/git-level and logical ones).

Reply:
sure, that would be a great fix because so many bugs are missed/nonsense happens because of that behavior, but this is def more important :)